zhusuan/distributions/univariate.py

```diff
@@ -746,13 +746,13 @@ def _get_batch_shape(self):
         return self.rate.get_shape()
 
     def _sample(self, n_samples):
-        if "random_poisson" in dir(tf):
+        try:
             # tf.random_poisson is implemented after v1.2
             samples = tf.random_poisson(self.rate, [n_samples],
                                         dtype=self.param_dtype)
             if self.param_dtype != self.dtype:
                 samples = tf.cast(samples, self.dtype)
-        else:
+        except AttributeError:
```

Comment:
only try: random_possion = tf.random_poisson is better since their may be other attribute error thrown out in the current block.